x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -493,7 +490,7 @@ public void testGetFieldFilterSecurityEnabled() throws Exception {
         assertTrue(fieldFilter.apply(randomAlphaOfLengthBetween(3, 6)).test("field_granted"));
         assertTrue(fieldFilter.apply(randomAlphaOfLengthBetween(3, 6)).test(randomAlphaOfLengthBetween(3, 10)));
         assertEquals(MapperPlugin.NOOP_FIELD_PREDICATE, fieldFilter.apply(randomAlphaOfLengthBetween(3, 10)));
-        expectThrows(IllegalStateException.class, () -> fieldFilter.apply("index_not_granted"));
+        assertEquals(MapperPlugin.NOOP_FIELD_PREDICATE, fieldFilter.apply("index_not_granted"));
```

Comment:
I think we should delete this assertion and related variables. It is no longer a valid case and rather confusing.

Reply:
Completely agree! I have pushed [0be3c98](https://github.com/elastic/elasticsearch/pull/90141/commits/0be3c9851bd9589a1e1e2554d808b61093c9cbc7) . It still checks the main code (which itself is lenient), but I think the test is less confusing now (though still checks the leniency).